builds.sh

```diff
@@ -160,11 +184,20 @@ for sourceId; do
 			buildIdJson="$(jq <<<"$buildIdParts" -c 'del(.resolvedParents)')" # see notes above (where buildIdParts is first defined)
 			buildId="$(sha256sum <<<"$buildIdJson" | cut -d' ' -f1)" # see notes above (where buildIdParts is first defined)
 			printf >&2 '%s\n' "$buildId"
+
 			img="$BASHBREW_STAGING_TEMPLATE"
 			img="${img//BUILD/$buildId}"
 			[ "$img" != "$BASHBREW_STAGING_TEMPLATE" ] # BUILD is required, for proper uniqueness (ARCH is optional)
 			img="${img//ARCH/$arch}"
-			resolved="$(_resolve "$img" "$arch")"
+
+			if [ -v "imageArchResolved["$img"]" ]; then
```

Comment:
```suggestion
			if [ -n "${imageArchResolved[$img]+x}" ]; then
```

Reply:
```suggestion
			if [ -n "${imageArchResolved[$img]:+x}" ]; then
```

If we also explicitly cache the null/empty/`{}` value.

Reply:
Done